notesapi/v1/tests/test_views.py

```diff
@@ -271,6 +325,74 @@ def test_read_all_no_query_param(self):
         response = self.client.get(reverse('api:v1:annotations'), self.headers)
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
 
+    @ddt.unpack
+    @ddt.data(
+        {'page': 1, 'annotations_per_page': 10, 'previous_page': None, 'next_page': 2},
+        {'page': 2, 'annotations_per_page': 10, 'previous_page': 1, 'next_page': 3},
+        {'page': 3, 'annotations_per_page': 3, 'previous_page': 2, 'next_page': None}
+    )
+    def test_pagination_multiple_pages(self, page, annotations_per_page, previous_page, next_page):
+        """
+        Verify that pagination info is correct when we have data spanned on multiple pages.
+        """
+        total_annotations = 23
+        for aid in range(total_annotations):
+            self._create_annotation(text=u'annotation {}'.format(aid))
+
+        response = self.get_annotations(query_parameters={'page': page})
+        self.verify_pagination_info(
+            response,
+            total_annotations=total_annotations,
+            num_pages=3,
+            annotations_per_page=annotations_per_page,
+            current_page=page,
+            previous_page=previous_page,
+            next_page=next_page
+        )
+
+    def test_pagination_single_page(self):
+        """
+        Verify that pagination info is correct when we have a single page of data.
+        """
+        total_annotations = 6
+        for aid in range(total_annotations):
+            self._create_annotation(text=u'annotation {}'.format(aid))
+
+        response = self.get_annotations()
+        self.verify_pagination_info(
+            response,
+            total_annotations=total_annotations,
+            num_pages=1,
+            annotations_per_page=total_annotations,
+            current_page=1,
+            previous_page=None,
+            next_page=None
+        )
+
+    @ddt.unpack
+    @ddt.data(
+        {'page_size': 2, 'annotations_per_page': 2, 'num_pages': 8, 'next_page': 2},
+        {'page_size': 15, 'annotations_per_page': 15, 'num_pages': 1, 'next_page': None},
+    )
+    def test_pagination_page_size(self, page_size, annotations_per_page, num_pages, next_page):
+        """
+        Verify that requests for different page_size returns correct pagination info.
+        """
+        total_annotations = 15
+        for aid in range(total_annotations):
+            self._create_annotation(text=u'annotation {}'.format(aid))
+
+        response = self.get_annotations(query_parameters={'page_size': page_size})
+        self.verify_pagination_info(
+            response,
+            total_annotations=total_annotations,
+            num_pages=num_pages,
+            annotations_per_page=annotations_per_page,
+            current_page=1,
+            previous_page=None,
+            next_page=next_page
+        )
```

Comment:
Can we create a helper method which has all these statements to avoid the duplications ? We can call that method from these tests.